jwst/skymatch/skymatch_step.py

```diff
@@ -216,3 +224,30 @@ def _set_sky_background(self, sky_image, library, step_status):
 
         dm.meta.cal_step.skymatch = step_status
         library.shelve(dm, index)
+
+
+    def _user_sky(self, library):
+        """Handle user-provided sky values for each image.
+        """
+
+        log.info(" ")
+        log.info("----  Using user-provided sky values for each image.")
+
+        if self.skylist is None:
+            raise ValueError('skymethod set to "user", but no sky values provided.')
+        if len(self.skylist) != len(library):
+            raise ValueError(f"Number of entries in skylist ({len(self.skylist)}) does not match "
+                             f"number of input images ({len(library)}).")
+
+        with library:
+            for i, model in enumerate(library):
+                sky = self.skylist[i]
+                model.meta.background.level = sky
+                model.meta.background.subtracted = self.subtract
+                model.meta.background.method = self.skymethod
+                if self.subtract:
+                    model.data -= sky
+                model.meta.cal_step.skymatch = "COMPLETE"
+                library.shelve(model)
```

Comment:
I suggest adding a log message so that users could debug issues or document what was done:
```suggestion
                self.log.info(
                    f"      - Image ID={i}. Setting sky background to user value: {sky:G} "
                    f"(file name: '{model.meta.filename}')."

                library.shelve(model)
```

Reply:
how does the new log message look @mcara ?

Reply:
It looks like your current log message doesn't report the value actually used for each file - I think that would be useful confirmation that the file was read in okay.

Reply:
missed this one, will add a debug message for each iteration in the loop

Reply:
This was added, resolving the conversation